docs/dev_guide.md

```diff
@@ -1,31 +1,149 @@
 # Dev Guide
 
-## Adding a new param in GCSFuse
+## Adding new features to GCSFuse
+
+**If you're interested in adding a new feature, please open a GitHub issue to
+discuss your proposal with the GCSFuse team first before making the
+contribution.**
+
+Rollout of any new feature should be done in 2 phases:
+
+**1. Experimental Phase:**
+
+* New features should initially be introduced behind an experimental flag. These
+  flags should be prefixed with `--experimental-` and marked as hidden in the
+  help output.
+* This signals to users that the feature is in development and may change or be
```

Comment:
If the feature is going to be hidden, then using the `--experimental` flag makes sense? I feel we should choose one approach: either create a hidden flag or a public flag with the `--experimental` prefix.

Reply:
I think it feels fine to have the prefix experimental as well. We can take the decision when to remove it based on whether it can be documented in public docs or not.